src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -910,9 +910,7 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
                 }
                 else
                 {
-                    // TODO-XArch-Bug https://github.com/dotnet/coreclr/issues/16329
-                    // using hardcoded instruction as workaround for inexact type conversions
-                    emit->emitIns_R_R(INS_mov_xmm2i, emitActualTypeSize(baseType), op1Reg, targetReg);
+                    emit->emitIns_R_R(ins, emitActualTypeSize(baseType), op1Reg, targetReg);
```

Comment:
Is this the only place where the signedness of the base type matters?

Reply:
It's the only place so-far. There are a few other intrinsics that will end up in the same boat, IIRC.

(there aren't many of them, however)

Reply:
So far it was the only place where the error was catched. The other place is an implementation of Crc32 intrinsic. I do expect to hit many more in near future.

Reply:
Ohh, I forgot to mention that in couple of cases that error was masked. For instance intrinsics taking as arguments `int`/`uint`/`long`/`ulong` will mask the problem since during codegen instructions for `int`/`long` will be selected and it just happened that they were the same as for `uint`/`ulong`. But it will not always be the case. This is true for  `Sse2.Add`, `Sse2.And`, `Sse2.AndNot` and many others.

Reply:
It migh be better to put together a list of intrinsics that have this problem. This approach deviates from the usual handling of unsigned operations in the JIT. That doesn't mean that it is bad but it's better to establish if it's worthwhile.

The typical JIT approach is to use specific unsigned operators (e.g. `GT_UDIV`) or the `GTF_UNSIGNED` flag (e.g. casts, relops).

Reply:
And it turns out that at least in this case this change is completely unnecessary. This is simply supposed to generate `MOVD` and the sign makes no difference whatsoever.

Reply:
It might just be best to add this next to the old `JITtype2varType(CorInfoType type)` in `ee_il_dll.hpp`
And add a comment to it explaining that the exact type including the sign dor INT and LONG is needed for the SIMD implementations.